task/sensor-api/src/lib.rs

```diff
@@ -8,26 +8,98 @@
 
 use userlib::*;
 
+use core::convert::TryFrom;
+
+use hubpack::SerializedSize;
+use serde::{Deserialize, Serialize};
 pub use task_sensor_types::*;
 
+#[derive(
+    zerocopy::AsBytes,
+    Copy,
+    Clone,
+    Debug,
+    Eq,
+    PartialEq,
+    Serialize,
+    SerializedSize,
+)]
+#[repr(C)]
+pub struct SensorId(u32);
+
+pub struct InvalidSensor;
+
+impl SensorId {
+    /// An invalid sensor ID.
+    ///
+    /// If this ID is actually passed to a call to a sensor API IPC operation,
+    /// the operation will return an error or reply-fault. However, this value
+    /// may be used as an initializer for `SensorId`s that will be set to a
+    /// valid value later.
+    pub const INVALID: Self = Self(u32::MAX);
```

Comment:
Does the existence of `SensorId::INVALID` break the "all `SensorId`s will be valid" invariant in a troublesome way?

I think the intention is to be checked at RPC boundaries, i.e. deserialization would fail if someone called `sensor.raw_reading(SensorId::INVALID)`, so it wouldn't get very far.

However, _within_ the `sensors` task, if someone happened to use `SensorId::INVALID`, it could be used to get out-of-bounds access to the data arrays.